package.json

```diff
@@ -437,6 +437,28 @@
             "markdownDescription": "Arguments to pass to SourceKit-LSP. Keys and values should be provided as individual entries in the list. e.g. `['--log-level', 'debug']`",
             "order": 2
           },
+          "swift.sourcekit-lsp.backgroundIndexing": {
+            "type": "boolean",
+            "default": false,
+            "markdownDescription": "Enable or disable background indexing. This option has no effect in Swift versions prior to 6.0.",
+            "order": 3
+          },
+          "swift.sourcekit-lsp.backgroundPreparationMode": {
+            "type": "string",
+            "enum": [
+              "build",
+              "noLazy",
+              "enabled"
+            ],
+            "default": "enabled",
+            "markdownDescription": "Determines how background indexing should prepare a target. Only used when Background Indexing is enabled. This option has no effect in Swift versions prior to 6.0.",
+            "enumDescriptions": [
+              "Build a target to prepare it",
+              "Prepare a target without generating object files but do not do lazy type checking and function body skipping",
+              "Prepare a target without generating object files and the like"
+            ],
+            "order": 4
+          },
```

Comment:
This setting is really there in case we discover issues with the `enabled` preparation mode but we can’t ship a new SourceKit-LSP version until the next release. For now, I don’t expect users to need to customize it and I’d remove the setting. Users can still customize it with a configuration file if needed and if this starts to become relevant, we can add it later.